core/src/broadcast_stage/broadcast_utils.rs

```diff
@@ -63,29 +63,33 @@ pub(super) fn recv_slot_entries(receiver: &Receiver<WorkingBankEntry>) -> Result
     let mut serialized_batch_byte_count = serialized_size(&entries)?;
 
     // Wait up to `ENTRY_COALESCE_DURATION` to try to coalesce entries into a 32 shred batch
-    let mut coalesce_deadline = Instant::now() + ENTRY_COALESCE_DURATION;
+    let mut coalesce_start = Instant::now();
     while last_tick_height != bank.max_tick_height()
         && serialized_batch_byte_count < target_serialized_batch_byte_count
     {
-        let (try_bank, (entry, tick_height)) = match receiver.recv_deadline(coalesce_deadline) {
-            Ok(working_bank_entry) => working_bank_entry,
-            Err(_) => break,
-        };
+        let (try_bank, (entry, tick_height)) =
+            match receiver.recv_deadline(coalesce_start + ENTRY_COALESCE_DURATION) {
+                Ok(working_bank_entry) => working_bank_entry,
+                Err(_) => break,
+            };
         // If the bank changed, that implies the previous slot was interrupted and we do not have to
         // broadcast its entries.
         if try_bank.slot() != bank.slot() {
             warn!("Broadcast for slot: {} interrupted", bank.slot());
             entries.clear();
             serialized_batch_byte_count = 8; // Vec len
             bank = try_bank;
-            coalesce_deadline = Instant::now() + ENTRY_COALESCE_DURATION;
+            coalesce_start = Instant::now();
         }
         last_tick_height = tick_height;
         let entry_bytes = serialized_size(&entry)?;
         serialized_batch_byte_count += entry_bytes;
         entries.push(entry);
         assert!(last_tick_height <= bank.max_tick_height());
     }
+    if let Some(stats) = process_shreds_stats {
+        stats.coalesce_elapsed_us += coalesce_start.elapsed().as_micros() as u64;
+    }
```

Comment:
This is not a good pattern to unnecessarily use an `Option` for the arg type adding an unnecessary branch to runtime. Plus it is not clear in this code why/when stats should be None.

Better to just pass `&mut ProcessShredsStats::default()` in tests and then ignore it, rather than adding extra overhead/verbosity to the actual runtime code.